### PR TITLE
feature file processor to support csf3 format

### PR DESCRIPTION
**Pull request by fa93hws**
there are both `yarn.lock` and `npm.lock`, which one should I use?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
     "enhanced-resolve": "^4.1.0",
     "esquery": "^1.3.1",
     "fast-glob": "^3.0.4",
-    "gherkin": "^7.0.3",
+    "gherkin": "^8.2.1",
```
Comment (fa93hws) [inline package.json:39]: `gherkin@7` doesn't come with a binary that's executable on m1

File: __tests__/fixtures/feature-storybook-ref/b.stories.tsx [hunk 2]
```diff
@@ -3,4 +3,11 @@
...
 
-storiesOf("b", module).add("Foo", () => null);
+export default {
+  title: 'b',
+  component: null,
+};
+
+export const Foo = {
+  render: () => null,
+}
```
Comment (bradzacher) [inline __tests__/fixtures/feature-storybook-ref/b.stories.tsx:13]: We'll need to keep supporting the old format as well, right?
they aren't migrating the entire codebase in one go I think?

File: src/processors/feature.ts [hunk 3]
```diff
@@ -129,6 +152,23 @@ export class FeatureFileProcessor extends TypeScriptFileProcessor {
...
+        if (FeatureFileProcessor.isObjectLiteralExpression(expression)) {
+          const property = expression.properties[0];
+          if (FeatureFileProcessor.isPropertyAssignment(property)) {
+            const { name, initializer } = property;
+            if (
+              FeatureFileProcessor.isIdentifier(name) &&
+              FeatureFileProcessor.isStringLiteral(initializer) &&
+              name.escapedText === CSF3_EXPORT_TITLE_FIELD
+            ) {
+              callback(initializer.text);
```
Comment (bradzacher) [inline src/processors/feature.ts:166]:
```suggestion
              callback(initializer.text);
              return;
```

can and should exit at this point as we've found the storybook name.

File: src/processors/feature.ts [hunk 4]
```diff
@@ -129,6 +152,23 @@ export class FeatureFileProcessor extends TypeScriptFileProcessor {
         return;
       }
 
+      if (FeatureFileProcessor.isExportAssignment(node)) {
+        const { expression } = node;
+        if (FeatureFileProcessor.isObjectLiteralExpression(expression)) {
```
Comment (bradzacher) [inline src/processors/feature.ts:157]:
```suggestion
        if (FeatureFileProcessor.isObjectLiteralExpression(expression) && expression.properties.length > 0) {
```

nit to guard against crashes for `export default {};`

**Review by bradzacher — APPROVED** [review]

File: __tests__/fixtures/feature-storybook-ref/b.stories.tsx (hunk 2 above)
Comment (fa93hws) [reply __tests__/fixtures/feature-storybook-ref/b.stories.tsx:13]: Yes, but we do have plenty other tests using the old format (`storiesOf`), such as `a.stories.tsx` and `c.stories.tsx`

**Review by bradzacher — APPROVED** [review]

**Pull request merged**